File: mongo/queries.js
// The current database to use.
use("yelp");

// Search for documents in the current collection.
db.getCollection("system.profile")
    .find(
        {
            op: "bulkWrite",
        },
        {
            millis: 1,
            ninserted: 1,
        }
    )
    .sort({
        /*
         * fieldA: 1 // ascending
         * fieldB: -1 // descending
         */
    });

use("yelp");
db.getCollection("system.profile").aggregate([
    {
        $match: {
            // op: "bulkWrite",
            op: "update",
        },
    },
    {
        $group: {
            _id: "$ns",
            millis: { $sum: "$millis" },
            nModified: { $sum: "$nModified" },
            //The total CPU time spent by a query operation in nanoseconds.
            //This field is only available on Linux systems.
            totalCpuNanos: { $sum: "$cpuNanos" },
        },
    },
]);

//GET VOTE COUNT
db.getCollection("reviews").aggregate([
    {
        $match: {
            _id: "IULD0OBwjh3Eoc0IotK1rA", //: "update"
        },
    },

    {
        $project: {
            count: { $size: "$votes" },
        },
    },
]);

//get average stars rating of business
db.getCollection("reviews").aggregate([
    {
        $match: {
            business_id: "AOdw-xjhc-ZRVhJzAgJxSw",
        },
    },
    {
        $group: {
            _id: "$business_id",
            avgStars: { $avg: "$stars" },
            voteCount: { $sum: 1 },
        },
    },
]);

//get all reviews for specific business
db.getCollection("reviews").find(
    {
        business_id: "AOdw-xjhc-ZRVhJzAgJxSw",
    },
    {
        business_id: 0,
    }
);

File: mongo/queries.py
import pymongo
import time


client = pymongo.MongoClient("mongodb://mongo:zaq1%40WSX@localhost:27017/")

db = client["yelp"]


start_time = time.time()

ids = ["gmjsEdUsKpj9Xxu6pdjH0g","e4Vwtrqf-wpJfwesgvdgxQ", "04UD14gamNjLY0IDYVhHJg"]

reviews = db.reviews

samples_count = 1000
for i in range(0, samples_count):
    reviews.find({
        "business_id": ids[i % len(ids)]
    })


total = time.time() - start_time
print("Fetch all reviews of bussiness")
print(f"Total: {total}, average: {total / samples_count}")

start_time = time.time()
for i in range(0, samples_count):
    reviews.aggregate([{
            "$match": {
                "business_id": ids[i % len(ids)],
            },
        },
        {
            "$group": {
                "_id": "$business_id",
                "avgStars": { "$avg": "$stars" },
                # "voteCount": { "$sum": 1 },
            },
        }
    ])

total = time.time() - start_time
print("Fetch average rating of bussiness")
print(f"Total: {total}, average: {total / samples_count}")



start_time = time.time()
for i in range(0, samples_count):
    reviews.count_documents(
        {"business_id": ids[i % len(ids)]}
    )


total = time.time() - start_time
print("Fetch review count of bussiness")
print(f"Total: {total}, average: {total / samples_count}")